addon/components/lt-body.js

```diff
@@ -266,32 +271,46 @@ export default Component.extend({
       let isSelected = row.get('selected');
       let currIndex = rows.indexOf(row);
       let prevIndex = this._prevSelectedIndex === -1 ? currIndex : this._prevSelectedIndex;
+      let otherAffectedRows = {
+          selected: Ember.A(),
+          expanded: Ember.A()
+      };
 
       this._currSelectedIndex = currIndex;
       this._prevSelectedIndex = prevIndex;
 
       if (canSelect) {
         if (e.shiftKey && multiSelect) {
-          rows.slice(Math.min(currIndex, prevIndex), Math.max(currIndex, prevIndex) + 1).forEach(r => r.set('selected', !isSelected));
+          rows.slice(Math.min(currIndex, prevIndex), Math.max(currIndex, prevIndex) + 1).forEach(r => {
+            if (!r.get('selected')) {
+                otherAffectedRows.selected.pushObject(r);
+            }
+
+            r.set('selected', !isSelected);
+          });
           this._prevSelectedIndex = currIndex;
         } else if ((!multiSelectRequiresKeyboard || (e.ctrlKey || e.metaKey)) && multiSelect) {
           row.toggleProperty('selected');
         } else {
+          otherAffectedRows.selected.pushObjects(this.get('table.selectedRows'));
           this.get('table.selectedRows').setEach('selected', false);
           row.set('selected', !isSelected);
 
           if (this.get('canExpand') && this.get('expandOnClick')) {
-            this.toggleExpandedRow(row);
+            otherAffectedRows.expanded = this.toggleExpandedRow(row);
           }
         }
         this._prevSelectedIndex = currIndex;
       } else {
         if (this.get('canExpand') && this.get('expandOnClick')) {
-          this.toggleExpandedRow(row);
+          otherAffectedRows.expanded = this.toggleExpandedRow(row);
         }
       }
 
-      callAction(this, 'onRowClick', ...arguments);
+      otherAffectedRows.selected = otherAffectedRows.selected.uniq().without(row);
```

Comment:
`affectedRows.selected = emberArray(affectedRows.selected).uniq().without(row)`

same for expanded